server/build_event_protocol/build_status_reporter/build_status_reporter.go

```diff
@@ -131,21 +131,23 @@ func (r *BuildStatusReporter) flushPayloadsIfWorkspaceLoaded(ctx context.Context
 			break
 		}
 		commitSHA := r.buildEventAccumulator.CommitSHA()
-		r.githubClient.CreateStatus(ctx, ownerRepo, commitSHA, payload)
+		r.githubClient.CreateStatus(ctx, ownerRepo, commitSHA, r.appendStatusNameSuffix(payload))
 	}
 
 	r.payloads = make([]*github.GithubStatusPayload, 0)
 }
 
-func (r *BuildStatusReporter) newGithubStatusPayload(name, URL, description string, state github.State) *github.GithubStatusPayload {
-	if r.statusNameSuffix != "" {
-		name = fmt.Sprintf("%s %s", name, r.statusNameSuffix)
+func (r *BuildStatusReporter) appendStatusNameSuffix(payload *github.GithubStatusPayload) *github.GithubStatusPayload {
+	if r.statusNameSuffix == "" {
+		return payload
 	}
-	return github.NewGithubStatusPayload(name, URL, description, state)
+	clone := *payload
```

Comment:
we should be careful doing this in general because if the struct changes to contain other stuff, the clone might not do what we think it should be doing.

If you don't need this, you could just drop the return argument of appendStatusNameSuffix and either mutate (or don't) payload in the method.

Reply:
I avoided mutating it because the status name is used as a map key and there is some code that relies on the map key matching the status name. Updated to create a new instance instead of copying via de-reference.